vizro-core/src/vizro/actions/_actions_utils.py

```diff
@@ -152,73 +160,76 @@ def _get_parametrized_config(target: ModelID, ctd_parameters: list[CallbackTrigg
                 selector_value = selector.options
 
         selector_value = _validate_selector_value_none(selector_value)
-        selector_actions = _get_component_actions(model_manager[ctd["id"]])
 
-        for action in selector_actions:
+        for action in _get_component_actions(model_manager[ctd["id"]]):
             if action.function._function.__name__ != "_parameter":
                 continue
 
-            action_targets = _create_target_arg_mapping(action.function["targets"])
-
-            if target not in action_targets:
-                continue
-
-            for action_targets_arg in action_targets[target]:
-                config = _update_nested_graph_properties(
-                    graph_config=config, dot_separated_string=action_targets_arg, value=selector_value
+            for dot_separated_string in _filter_dot_separated_strings(action.function["targets"], target, data_frame):
+                config = _update_nested_figure_properties(
+                    figure_config=config, dot_separated_string=dot_separated_string, value=selector_value
                 )
 
     return config
 
 
 # Helper functions used in pre-defined actions ----
-def _get_targets_data_and_config(
+def _apply_filters(
+    data: pd.DataFrame,
     ctds_filter: list[CallbackTriggerDict],
     ctds_filter_interaction: list[dict[str, CallbackTriggerDict]],
-    ctds_parameters: list[CallbackTriggerDict],
-    targets: list[ModelID],
+    target: ModelID,
 ):
-    all_filtered_data = {}
-    all_parameterized_config = {}
+    # Takes in just one target, so dataframe is filtered repeatedly for every target that uses it.
+    # Potentially this could be de-duplicated but it's not so important since filtering is a relatively fast
+    # operation (compared to data loading).
+    filtered_data = _apply_control_filters(data_frame=data, ctds_filters=ctds_filter, target=target)
+    filtered_data = _apply_filter_interaction(
```

Comment:
I know its a big change, but maybe we can do it now - it seems more in line with the term `control_filters` and I think would help people understand that they are on equal footing

Reply:
Yeah I considered this before and decided not to because:
* the public-facing function is `filter_interaction` and cannot be renamed
* eventually all the `filter_interaction` stuff will be removed and we'll just have `_apply_filters`

So I think the easiest way to make this consistent for now is to rename `control_filters` to `filter_controls`.

Reply:
[1509408](https://github.com/mckinsey/vizro/pull/857/commits/150940846ae4df7287c6e77a7cb92407199ae1fc)